### ISSUE TITLE
Problem with rast-process-genome: Gateway Time-out

### ISSUE THREAD

**Issue by lzaramela** [TheSEED/RASTtk-Distribution#9]
Hi,
I am trying to annotate multiple genomes and I am getting a Gateway Time-out error. Sometimes after several attempts, it gets done, but in many cases, an empty file is created. See error below:

rast-process-genome < genome1.renamed.gto> genome1.renamed.gto.gto2
2019-04-04 09:03:06: Request failed with code=504 msg=Gateway Time-out, sleeping 1 and retrying
2019-04-04 09:04:13: Request failed with code=504 msg=Gateway Time-out, sleeping 2 and retrying
2019-04-04 09:05:19: Request failed with code=504 msg=Gateway Time-out, sleeping 5 and retrying
2019-04-04 09:06:27: Request failed with code=504 msg=Gateway Time-out, sleeping 10 and retrying
2019-04-04 09:11:51: Request succeeded after 4 retries

**Comment by jdwinkler-lanzatech** [comment]
I'm also having the same problem.

**Comment by cssanchez23** [comment]
I am also having this problem, how did you all resolve this issue?

**Comment by WallyL** [comment]
I'm having this same issue today as well.  So I guess no one is going to address this?

**Comment by olsonanl** [comment]
Let me check the service. Are you running single genomes or did you submit a batch?

**Comment by WallyL** [comment]
I'm just running single genomes

**Comment by WallyL** [comment]
Two jobs have both failed after 8 requests (restart after previous failures).  I'll kill it and wait until I hear something back from you.

**Comment by olsonanl** [comment]
Can you tell me your IP address so I can track errors in the service logs? I'm seeing running jobs on the service currently; how large are the genomes you are running (both in DNA size and number of contigs)?

**Comment by WallyL** [comment]
128.192.15.224
Both 5 MB, both single contigs

**Comment by WallyL** [comment]
Mb

**Comment by olsonanl** [comment]
I'm having the machine the service is running on rebooted; that should help. There were a bunch of stuck connections it looks like.

**Comment by WallyL** [comment]
Okay, thank you for getting on this.  I've been using the PATRIC/RASTtk service for a few years now and this is the first time I've run into this issue.  Perhaps it's getting too popular!
Walt

**Comment by WallyL** [comment]
I just tried toi run again.  Same result....
2020-01-27 14:11:54: Request failed with code=502 msg=Bad Gateway, sleeping 1 and retrying

**Comment by olsonanl** [comment]
Yes, issue with the machine coming up. Waiting on systems folks.

**Comment by olsonanl** [comment]
Have you used the PATRIC genome annotation service for analyzing the genomes? It is more robust than the rasttk version (it is much more heavily used).

**Comment by WallyL** [comment]
I haven't simply because I've been using the rasttk pipeline for a long time and I have multiple gto and export scripts that I plug and play.  Looks like I need to upgrade...

**Comment by olsonanl** [comment]
We're back. I also allocated more resources to the service. We brought up a new scheduler for the rest of the production services so at this point the host for the rasttk annotation service isn't running other things.

**Comment by WallyL** [comment]
Okay, the export files I was trying to get worked.  Thanks!

Quick question regarding P3.  Below is an example script I typically run for annotating genomes that I've assembled followed by export in different formats:

> #!/bin/bash
rast-create-genome --scientific-name "Aeromonas hydrophila" --genetic-code 11 --domain Bacteria --contigs Ah_43409_flye_pbio.fa > AH43409.gto
rast-process-genome < AH43409.gto > AH43409.gto2
rast-annotate-proteins-kmer-v2	<	AH43409.gto2	>	AH43409.gto3
rast-annotate-proteins-kmer-v1 -H	<	AH43409.gto3	>	AH43409.gto4
rast-annotate-proteins-similarity -H	<	AH43409.gto4	>	AH43409.gto5
rast-annotate-special-proteins	<	AH43409.gto5	>	AH43409.gto6
rast-annotate-families-patric	<	AH43409.gto6	>	AH43409.gto7
rast-call-features-prophage-phispy < AH43409.gto7 > AH43409.gto8 
rast-export-genome genbank < AH43409.gto8 > AH43409.gbk
rast-export-genome genbank_merged <AH43409.gto8> AH43409_merged.gbk
rast-export-genome gff < AH43409.gto8 > AH43409.gff3
rast-export-genome feature_data  <AH43409.gto8> AH43409.features.txt
rast-export-genome spreadsheet_txt < AH43409.gto8 > AH43409.txt
rast-export-genome spreadsheet_xls < AH43409.gto8 > AH43409.xls
rast-export-genome protein_fasta <AH43409.gto8> AH43409.pep.fa
rast-export-genome patric_specialty_genes <AH43409.gto8> AH43409.patric_specialty.txt
rast-export-genome feature_dna <AH43409.gto8> AH43409.feature.fa
rast-export-genome patric_features <AH43409.gto8> AH43409.patric_features.txt


From reading the documentation on P3 here: [https://docs.patricbrc.org/cli_tutorial/cli_getting_started.html](url) it looks like you can cumulatively add on to P3 outputs similar to adding on to gtos in rastk and then export, or select fields that you want to export, via p3-echo function and output is a tabbed file.  Is this correct?  So there is no analogous export feature to generate, for example .gff, .gbk or .xls outputs as in rasttk?  I think this is why I didn't change over when I first saw this.  Rasttk my be overall less flexible than P3, but, to me, unless I'm missing something, it seems a bit simpler to navigate & invoke.

**Comment by olsonanl** [comment]
Ah, you're looking at the lower level commands. If you are looking at genome annotation, the standard PATRIC analysis pipeline will run what you list above, and does the exports. I'd advise running a job from the P3 website and take a look at the generated output. You can use p3-submit-genome-annotation to automate the submission process.

**Comment by cizydorczyk** [comment]
Having a similar issue today: 

`2021-11-08 13:44:22: Request failed with code=500 msg=read timeout, sleeping 1 and retrying`

Been trying since this morning. Had a few successfully complete, and now whichever sample I try, the above errors comes up.

Annotating single genomes at a time.
